addons/account/models/account_move_line.py

```diff
@@ -1631,16 +1644,19 @@ def write(self, vals):
 
             # Check the lock date.
             if line.parent_state == 'posted' and any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in protected_fields['fiscal']):
-                line.move_id._check_fiscalyear_lock_date()
+                fiscal_moves |= line.move_id
```

Comment:
This operation is `O(n)` (leading to `O(n²)` for the loop) because recordsets are immutable, would be better to append on a list of ids instead and re-browse at the end